### PR TITLE
CHEF-207 Validate and improve document of Chef Backend to Automate HA migration flow

### PR DESCRIPTION
**Pull request by shanmugapriya-tr**
### :nut_and_bolt: Description: What code changed, and why?

Updated the public documentation for Migration of Chef Infra Server/Chef Backend to Automate HA

### :chains: Related Resources

https://chefio.atlassian.net/browse/CHEF-207 

### :athletic_shoe: How to Build and Test the Change

Please check the netlify preview in the PR

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-04-04 at 4 00 50 PM
Screenshot 2023-04-04 at 4 00 26 PM

### CODE REVIEW

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 1]
```diff
@@ -17,276 +15,299 @@ gh_repo = "automate"
...
+`knife-ec-backup` utility backups and restores the data in an Enterprise Chef Server installation, preserving the data in an intermediate, editable text format. It is similar to the knife download, uploads commands, and uses the same underlying libraries. It includes workarounds for unsupported objects by the tools and various Server API deficiencies. The goal is to improve knife download, knife upload, and the Chef Infra Server API to deprecate the tool.
 
 {{< note >}}
 
-- The migration procedure is tested on Chef Server version 14+.
-- The migration procedure is possible above Chef Backend version 2.1.0.
+- The migration procedure is tested on Chef Server versions 14+ and 15+.
+- The migration procedure is tested and is possible on the Chef Backend version above 3.1.1.
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:38]: We can change it back to 2.1.0. We hav tested on that also now.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 2]
```diff
@@ -17,276 +15,299 @@ gh_repo = "automate"
...
-For example:
+    - `with-user-sql` is required to handle user passwords and ensure user-specific association groups are not duplicated.
+    - `--with-key-sql` is to handle cases where customers have users with multiple pem keys associated with their user or clients. The current chef-server API only dumps the default key. Sometimes, users will generate and assign additional keys to give additional users access to an account but still be able to lock them out later without removing everyone's access.
 
-```cmd
-    hab pkg exec chef/knife-ec-backup knife ec backup backup_$(date '+%Y%m%d%H%M%s') --webui-key /etc/opscode/webui_priv.pem -s https://chef.io`.
-```
+    ```cmd
+        hab pkg exec chef/knife-ec-backup knife ec backup backup_$(date '+%Y%m%d%H%M%s') --webui-key /etc/opscode/webui_priv.pem -s <chef server
+    ```
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:83]: it shpuld be <chef server url> ?

**Review by YashviJain01 — APPROVED** [review]

**Review by daveaugustus — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 3]
```diff
@@ -17,276 +15,299 @@ gh_repo = "automate"
...
 
-## In place Migration (Chef Backend to Automate HA)
+## Steps to validate if Migration is successful
 
-As part of this scenario, customer will migrate from chef-backend (5 machines) to Automate HA in-place, i.e. Automate HA will be deployed in those 5 machines only where Chef-backend is running. One extra bastion node will be required which will be managing the deployment of Automate HA on the chef backend infrastructure.
+- Execute the below command from the old server where a knife is installed and from the new server where the knife is installed :
 
-{{< note >}}
+    ```cmd
+        bash infra_server_objects_count_collector.sh -S test -K Key -F Filename
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:124]: Where to find the file, we have to mention this.

File: components/docs-chef-io/content/automate/ha_troubleshooting.md [hunk 4]
```diff
@@ -185,15 +164,85 @@ terraform taint module.bootstrap_automate.null_resource.automate_post[2]
...
+
+### Issue: knife SSL cert while setting up workstation during migration
+
+In the case of HA setup, while doing `knife ssl fetch`, if the certificate fetched is for the domain *.chefdemo.net, follow the below steps when you run `knife ssl check`.
+
+#### Solution
+
+- Go to route 53 chefdemo.net
+- Create record as recordname.eng.chefdemo.net
+- CNAME with value HA URL: ec2-3-108-254-0.ap-south-1.compute.amazonaws.com
```
Comment (YashviJain01) [inline components/docs-chef-io/content/automate/ha_troubleshooting.md:247]: don't put this ex2 instance name here.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 1 above)
Comment (kalroy) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:38]: Let's move it back to 2.1.0 then.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 5]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
-- The migration procedure is tested on Chef Server version 14+.
-- The migration procedure is possible above Chef Backend version 2.1.0.
+- The migration procedure is tested on Chef Server versions 14+ and 15+.
+- The migration procedure is tested and is possible on the Chef Backend version above 3.1.1.
 
 {{< /note >}}
 
+## Prerequisites
+
+Check the [Prerequisites](/automate/ha_aws_deployment_prerequisites/#migration) page before migrating.
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:44]: We should mention for for both Onprem as well as AWS

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 6]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
+
 ## Backup the Existing Chef Infra Server or Chef Backend Data
 
-1.   Execute below command to install Habitat:
+1. Execute the below command to install Habitat:
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:48]: Do we need to install tab while running `knife-ec-backup` on a Chef Server or Instance?

Comment (kalroy) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:48]: This doc talks about installing using `ruby gem`
https://github.com/chef/knife-ec-backup#chef-infra-server-install-recommended

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 7]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
-```cmd
-    hab pkg exec chef/knife-ec-backup knife tidy server report --node-threshold 60 -s https://chef.io -u pivotal -k /etc/opscode/pivotal.pem
-```
+    - `pivotal` is the name of the user
+    - `path of pivotal` is where the user's pem file is stored.
+    - `node-threshold NUM_DAYS` is the maximum number of days since the last checking before a node is considered stale.
 
-4.   Execute the below command to initiate a backup of your Chef Server data. 
-    In this command:
+    ```cmd
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:66]: Opinion: the command should go above before the details of the parameters.
cc: @dishanktiwari2501

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 8]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
+        hab pkg exec chef/knife-ec-backup knife tidy server report --node-threshold 60 -s https://chef.io -u pivotal -k /etc/opscode/pivotal.pem
+    ```
 
-```cmd
-    hab pkg exec chef/knife-ec-backup knife ec backup backup_$(date '+%Y%m%d%H%M%s') --webui-key /etc/opscode/webui_priv.pem -s <chef server
-```
+4. Execute the below command to initiate a backup of your Chef Server data. In this command:
 
-For example:
+    - `with-user-sql` is required to handle user passwords and ensure user-specific association groups are not duplicated.
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:78]:
```suggestion
    - `--with-user-sql` is required to handle user passwords and ensure user-specific association groups are not duplicated.
```

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 9]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
-```
+    ```cmd
+        hab pkg exec chef/knife-ec-backup knife ec restore <path/to/directory/backup_file> -yes --concurrency 1 --webui-key /hab/svc/automate-cs-oc-erchef/data/webui\_priv.pem --purge -c /hab/pkgs/chef/chef-server-ctl/*/*/omnibus-ctl/spec/fixtures/pivotal.rb
+    ```
 
-## In place Migration (Chef Backend to Automate HA)
+## Steps to validate if Migration is successful
 
-As part of this scenario, customer will migrate from chef-backend (5 machines) to Automate HA in-place, i.e. Automate HA will be deployed in those 5 machines only where Chef-backend is running. One extra bastion node will be required which will be managing the deployment of Automate HA on the chef backend infrastructure.
+- Execute the below command from the old server where a knife is installed and from the new server where the knife is installed :
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:121]: Let's be more explicit and split this statements into two steps:
- Fetching for older server. 
  <commad>
- Fetching from HA
<command>
-  Check the difference

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 10]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
 
-{{< note >}}
+    ```cmd
+        curl https://raw.githubusercontent.com/chef/automate/main/dev/infra_server_objects_count_collector.sh -o infra_server_objects_count_collector.sh
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:124]: These are two different commands

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 11]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
 ```
 
-10. Clean up the old packages from chef-backend (like Elasticsearch and postgres)
-
-11. [Doc: Restore Backed Up Data to Chef Automate HA](/automate/ha_chef_backend_to_automate_ha/#restore-backed-up-data-to-chef-automate-ha)
+10. Clean up the old packages from the chef-backend (like Elasticsearch and Postgres)
 
+11. [Restore Backed Up Data to Chef Automate HA](/automate/ha_chef_backend_to_automate_ha/#restore-backed-up-data-to-chef-automate-ha)
 
 ## Using Automate HA for Chef-Backend user
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:270]: This title reads confusing for me. @dishanktiwari2501

Comment (kalroy) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:270]: We need to add a section to modify existing nodes to use the newly deployed and migrated Chef Automate HA instance. 
@YashviJain01  @shanmugapriya-tr

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 12]
```diff
@@ -17,276 +15,334 @@ gh_repo = "automate"
...
+- `-K` is the path of pivotal.pem file
+- `-F` is the path to store the output file
+- Now run the below command to check the differences between the old and new data. Ideally, there should be no differences if the migration was done successfully.
 
-{{< /note >}}
+    ```cmd
+        diff old_server_file new_server_file
+    ```
+
+## In-place Migration (Chef Backend to Automate HA)
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:137]: We should add a section to migrate Chef Server/Backend with S3 to Automate HA. We can talk about two sub scenarios here:
- If we want to use the same S3 bucket
- If we want to use a different S3 bucket
The section should talk about steps to configure and migrate as well the implications of these two approaches

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/ha_troubleshooting.md [hunk 13]
```diff
@@ -20,45 +18,38 @@ This page explains the frequently encountered issues in Chef Automate High Avail
...
-- Run the command on Automate HA chef-server node `journalctl --follow --unit chef-automate`
-  if getting an 500 internal server error with data-collector endpoint, it means that
-  Chef Infra Server not able to communicate to the Chef Automate data-collector endpoint
-  
-  ssh to the Automate HA Chef Infra Server, and get token and automate-lb-url from the config.
-  run `chef-automate config show` go get the config.
-  
+### Post Automate HA deployment if the chef-server service is in a critical state
+
+- Run the command on Automate HA chef-server node `journalctl --follow --unit chef-automate` if getting a 500 internal server error with the data-collector endpoint, it means that Chef Infra Server is not able to communicate to the Chef Automate data-collector endpoint ssh to the Automate HA Chef Infra Server, and get the token and automate-lb-url from the config. Run `chef-automate config show` to get the config.
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_troubleshooting.md:23]: This statement requires review. Missing "." and ";". @dishanktiwari2501

File: components/docs-chef-io/content/automate/ha_troubleshooting.md [hunk 14]
```diff
@@ -185,15 +164,86 @@ terraform taint module.bootstrap_automate.null_resource.automate_post[2]
 While trying to deploy or upgrade Chef Automate HA, if the command did not exit gracefully, the possible error looks like as shown below:
 
 ```bash
-      Exiting as execution process completed
-UnknownError: Unable to determine error cause: Command did not exit gracefully
+      Exiting as the execution process completed
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_troubleshooting.md:167]: Is this actually the error or modified for the grammatical correction? If so, we should stick to the error the command throws

File: components/docs-chef-io/content/automate/ha_troubleshooting.md [hunk 15]
```diff
@@ -185,15 +164,86 @@ terraform taint module.bootstrap_automate.null_resource.automate_post[2]
...
-- In logs check for which module it prompted an error.
+- In logs, check for which module prompted an error.
 - SSH into the node and check if the `sysctl` utility is available by running `sysctl -a`.
-- If `Command 'sysctl' not found`, try installing the pkg. 
\ No newline at end of file
+- If `Command 'sysctl' not found`, try installing the pkg.
+
+### Issue: knife-ec-restore during migration: Failed to restore synchronous operations
+
+- While running the restore command, If you are getting this error in logs: Failed to restore synchronous operations,  follow either of the methods below.
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_troubleshooting.md:182]:
```suggestion
- While running the restore command, if you are getting this error in logs: 
`Failed to restore synchronous operations` 
 follow either of the methods below.
```

File: components/docs-chef-io/content/automate/ha_troubleshooting.md [hunk 16]
```diff
@@ -185,15 +164,86 @@ terraform taint module.bootstrap_automate.null_resource.automate_post[2]
...
+done
+```
+
+#### Solution 2
+
+- Log in to the OpenSearch dashboard
+- Run this query: GET _snapshot/_all to get all the snapshots.
+- Delete all the snapshots using this query: DELETE _snapshot/<snapshot name>
+
+  For example: DELETE _snapshot/ chef-automate-es6-event-feed-service
```
Comment (kalroy) [inline components/docs-chef-io/content/automate/ha_troubleshooting.md:237]:
```suggestion
  For example: DELETE _snapshot/chef-automate-es6-event-feed-service
```

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 11 above)
Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:318]: The steps are mentioned below. DO we need any change in that @kalroy

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 6 above)
Comment (shanmugapriya-tr) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:52]: @kalroy We have tested migration by installing hab and then knife-ec-backup package only

File: components/docs-chef-io/content/automate/ha_troubleshooting.md (hunk 14 above)
Comment (shanmugapriya-tr) [reply components/docs-chef-io/content/automate/ha_troubleshooting.md:167]: @kalroy @dishanktiwari2501 - This seems to be corrected for grammatical error. Reverted this.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 12 above)
Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:160]: We have added a section. Can you please have a look. @kalroy

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 17]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
+    ```bash
+        hab pkg exec chef/knife-ec-backup knife tidy server clean --backup-path /path/to/an-ec-backup
+    ```
 
-## Restore Backed Up Data to Chef Automate HA
+5. Execute the below command to copy the backup directory to the Automate HA Chef Server.
 
--   Execute the below command to install the habitat package for `knife-ec-backup`
+    ```cmd
+        scp -i /path/to/key backup\_$(date '+%Y%m%d%H%M%s') user@host:/home/user
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:102]: Should we instead write <path of your backup file> something like that rather than using backup\_$(date '+%Y%m%d%H%M%s')

Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md]: Added >backup-file> rather than any particular date with it.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 18]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
+    ```bash
+        hab pkg exec chef/knife-ec-backup knife tidy server clean --backup-path /path/to/an-ec-backup
+    ```
 
-## Restore Backed Up Data to Chef Automate HA
+5. Execute the below command to copy the backup directory to the Automate HA Chef Server.
 
--   Execute the below command to install the habitat package for `knife-ec-backup`
+    ```cmd
+        scp -i /path/to/key <backup-file> user@host:/home/user
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:102]: backup file or path to backup file. Just like we are writing for key

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 19]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
-## In place Migration (Chef Backend to Automate HA)
+Before restoring the backup on the Automate HA Chef Server, If S3 storage is required for cookbooks. Please [Configure S3](/automate/chef_infra_external_cookbooks_in_chef_automate/)
 
-As part of this scenario, customer will migrate from chef-backend (5 machines) to Automate HA in-place, i.e. Automate HA will be deployed in those 5 machines only where Chef-backend is running. One extra bastion node will be required which will be managing the deployment of Automate HA on the chef backend infrastructure.
 
 {{< note >}}
 
-This will require downtime, so plan accordingly. A reduced performance should be expected with this. 
+- If using same bucket for Automate HA , new files will be uploaded for cookbooks. There will not be any impact on the old files.
+- If using the a new bucket fot Auomate HA, new files will be uplaoded for cookbooks
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:116]: It has few grammatical error
1. There is a space between comma and HA
2. fot -> for
3 auomate -> automate 
4 uplaoded -> uploaded

The sentence <new files will be uploaded for cookbooks.> is not clear to me

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 18 above)
Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:102]: done.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 20]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
 
 {{< /note >}}
 
+The migration flow has been tested on following scenaiors 
+
+- If the cookbooks were not stored in s3, at time of migration it can be resorted in S3.
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:122]: at the time of migration ?

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 21]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
+- If the cookbooks were stored in s3, at time of migration it can either be restored in s3 or postgres.
+
+## Restore Backed Up Data to Chef Automate HA
+
+- Execute the below command to install the habitat package for `knife-ec-backup`
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:127]: Execute the below command where ? I mean should we mention on which machine user should run this command

Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:127]: I think thats not required. As it can be run anywhere.

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 22]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
+        hab pkg exec chef/knife-ec-backup knife ec restore <path/to/directory/backup_file> -yes --concurrency 1 --webui-key /hab/svc/automate-cs-oc-erchef/data/webui\_priv.pem --purge -c /hab/pkgs/chef/chef-server-ctl/*/*/omnibus-ctl/spec/fixtures/pivotal.rb
+    ```
+
+## Steps to validate if Migration is successful
+
+- Execute the below commands from the old server where the knife is installed :
+
+    ```cmd
+        curl https://raw.githubusercontent.com/chef/automate/main/dev/infra_server_objects_count_collector.sh -o infra_server_objects_count_collector.sh
+        bash infra_server_objects_count_collector.sh -S test -K Key -F Filename
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:145]: I think instead of test we can write <chef-server-url> to make it consistent like other places
Same for key. I think it should be <path to key>

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 23]
```diff
@@ -17,276 +15,354 @@ gh_repo = "automate"
...
-- Take the backup of the system to avoid the data loss.
+- Set up your workstation based on the newly created Automate-HA's chef-server. It is only needed if you have set up the workstation earlier.
+- This in-place migration works only when cookbooks are stored in a database. This does not support use-case, where cookbooks are stored in the filesystem.
+
 {{< /note >}}
 
-1. ssh to all the backend nodes of chef-backend and run
+{{< note >}}
+
+- In order to validate the In-place migration, please run the validation script before starting backup and restore and then once the below steps from 1 to 11 are done successfully, run it again.
```
Comment (rishabhjhs) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:175]: Should we make 1 to 11 are done 
to 
1 to 11 ran

**Review by dishanktiwari2501 — APPROVED** [review]
Have made the possible changes directly to the docs.

**Review by rishabhjhs — APPROVED** [review]

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md (hunk 10 above)
Comment (YashviJain01) [reply components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:141]: done

**Review by kalroy — APPROVED** [review]

**Review by jayvikramsharma1 — APPROVED** [review]

**Review by vivekshankar1 — APPROVED** [review]

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 24]
```diff
@@ -17,276 +15,368 @@ gh_repo = "automate"
...
 
 {{< warning >}}
+
 - Customers using only **Standalone Chef Infra Server** or **Chef Backend** are advised to follow this migration guidance. Customers using **Chef Manage** or **Private Chef Supermarket** with Chef Backend should not migrate with this.
-- Also, for the customers using a standalone Chef Infra Server, cookbooks should be in the database but not in either the file system or S3.
-- Automate HA do not support the super market authentication with chef-server users credentials. 
-- Post Migration Customer can not login with chef-server users to Supermarket. 
+- Also, for the customers using a standalone Chef Infra Server, cookbooks should be in the database or S3 but not in the file system.
+- Automate HA does not support supermarket authentication with chef-server user credentials.
+- Post Migration Customer can not log in with chef-server users to Supermarket.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:22]:
```suggestion
- Automate HA does not support supermarket authentication with chef-server user credentials. Post migration to Automate HA those customer users will not be able to log in with chef-server user credentials to their Supermarket.
```

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 25]
```diff
@@ -17,276 +15,368 @@ gh_repo = "automate"
...
+
 {{< /warning >}}
 
 This page explains the procedure to migrate the existing Standalone Chef Infra Server or Chef Backend data to the newly deployed Chef Automate HA. This migration involves two steps:
 
--   Back up the data from an existing Chef Infra Server or Chef Backend via `knife-ec-backup`.
--   Restore the backed-up data to the newly deployed Chef Automate HA environment via `knife-ec-restore`.
+- Back up the data from an existing Chef Infra Server or Chef Backend via `knife-ec-backup`.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:28]:
```suggestion
- Backup the data from an existing Chef Infra Server or Chef Backend via `knife-ec-backup`.
```

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 26]
```diff
@@ -17,276 +15,368 @@ gh_repo = "automate"
...
-
-`knife-ec-backup` utility backups and restores the data in an Enterprise Chef Server installation, preserving the data in an intermediate, editable text format. It is similar to the knife download and knife upload commands and uses the same underlying libraries. It includes workarounds for unsupported objects by the tools and various Server API deficiencies. The goal is to improve knife download, knife upload, and the Chef Infra Server API to deprecate the tool.
+Take a backup using the `knife-ec-backup` utility and move the backup folder to the newly deployed Chef Server. Later, restore using the same utility. The backup migrates all the cookbooks, users, data bags, policies, and organizations. `knife-ec-backup` utility backups and restores the data in an Enterprise Chef Server installation, preserving the data in an intermediate, editable text format. It is similar to the knife download, uploads commands, and uses the same underlying libraries. It includes workarounds for unsupported objects by the tools and various Server API deficiencies. The goal is to improve knife download, knife upload, and the Chef Infra Server API to deprecate the tool.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:31]: Why is this needed ?
`It is similar to the knife download, uploads commands, and uses the same underlying libraries. It includes workarounds for unsupported objects by the tools and various Server API deficiencies. The goal is to improve knife download, knife upload, and the Chef Infra Server API to deprecate the tool.`
I feel we can remove this.
```suggestion
Take a backup using the `knife-ec-backup` utility and move the backup folder to the newly deployed Chef Server. Later, restore using the same utility. The backup migrates all the cookbooks, users, data bags, policies, and organizations. `knife-ec-backup` utility backups and restores the data in an Enterprise Chef Server installation, preserving the data in an intermediate, in editable text format.
```

File: components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md [hunk 27]
```diff
@@ -17,276 +15,368 @@ gh_repo = "automate"
...
 
 {{< note >}}
 
-- The migration procedure is tested on Chef Server version 14+.
-- The migration procedure is possible above Chef Backend version 2.1.0.
+- The migration procedure is tested on Chef Server versions 14+ and 15+.
+- The migration procedure is tested and is possible on the Chef Backend version above 2.1.0.
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/ha_chef_backend_to_automate_ha.md:36]: I think @kalroy  mentioned we wanted to change the min. supported version to 3.x. 
@kalroy  please confirm on this.
